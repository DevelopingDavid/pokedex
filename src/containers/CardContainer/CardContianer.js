import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from '../../components/Card/Card';
import PropTypes from 'prop-types';

export class CardContainer extends Component {
  render() {
    const { pokemon } = this.props;
    return (
      <section className='cards-container'>
        {pokemon.length > 0 &&
          pokemon.map(pokemon => {
            return <Card key={pokemon.name} pokemonInfo={pokemon} />
          })}
      </section>
    )
  }
}

export const mapStateToProps = (state) => ({
  pokemon: state.pokemon
});

export default connect(mapStateToProps)(CardContainer);

CardContainer.propTypes = {
  Pokemon: PropTypes.array
}