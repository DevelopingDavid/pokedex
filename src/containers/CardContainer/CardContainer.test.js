import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer } from './CardContianer';

const mockPokemon = [{
  name: 'charmander',
  types: [{ slot: 2, type: { name: 'fire' } }]
}];

describe('CardContainer', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <CardContainer pokemon={mockPokemon}/>
    )
  });

  it('should match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});