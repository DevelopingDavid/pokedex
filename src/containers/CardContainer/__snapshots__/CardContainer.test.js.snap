// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match Snapshot 1`] = `
<section
  className="cards-container"
>
  <Card
    key="charmander"
    pokemonInfo={
      Object {
        "name": "charmander",
        "types": Array [
          Object {
            "slot": 2,
            "type": Object {
              "name": "fire",
            },
          },
        ],
      }
    }
  />
</section>
`;
